Plugins/NativeEngine/Source/NativeEngine.cpp

```diff
@@ -1410,6 +1417,9 @@ namespace Babylon
 
         if (m_boundVertexArray != nullptr)
         {
+            if (!m_boundVertexArray->IsValid())
+                return;
```

Comment:
What happens if DrawInternal is called without the vertex array binding the buffers?

Reply:
In the cases I tested nothing bad seams to happen but I think this might be render API specific. Moreover, I think if the shader is calling this with m_boundVertexArray not been null it should be expecting it to be bind, so I didn't want to fall into undefined behavior.

Reply:
As long as it doesn't crash, I'm okay with undefined behavior here since it can't draw it correctly regardless. We already can get into a state where the vertex array isn't bound and that's normal. If we are okay with this, then you should be able to simplify the code a bit without exposing IsValid.